op-test-sequencer/sequencer/backend/work/builders/standardbuilder/builder.go

```diff
@@ -62,35 +63,62 @@ func NewBuilder(id seqtypes.BuilderID,
 	}
 }
 
-func (b *Builder) NewJob(ctx context.Context, opts *seqtypes.BuildOpts) (work.BuildJob, error) {
+func (b *Builder) NewJob(ctx context.Context, opts seqtypes.BuildOpts) (work.BuildJob, error) {
+	b.log.Debug("Builder NewJob request", "opts", opts)
+
 	parentRef, err := b.l2.L2BlockRefByHash(ctx, opts.Parent)
 	if err != nil {
 		return nil, fmt.Errorf("failed to retrieve parent-block: %w", err)
 	}
-	l1OriginRef, err := b.l1.L1BlockRefByHash(ctx, *opts.L1Origin)
-	if err != nil {
-		return nil, fmt.Errorf("failed to retrieve L1 origin: %w", err)
+	b.log.Debug("Builder NewJob fetched parentRef", "ref", parentRef)
+
+	var l1OriginRefBlkID eth.BlockID
+
+	if opts.L1Origin != nil {
+
+		b.log.Debug("About to get block ref by hash for l1 origin", "opts", opts)
+		l1OriginRef, err := b.l1.L1BlockRefByHash(ctx, *opts.L1Origin)
+		if err != nil {
+			return nil, fmt.Errorf("failed to retrieve L1 origin: %w", err)
+		}
+		l1OriginRefBlkID = l1OriginRef.ID()
+
+		b.log.Debug("Builder NewJob fetched l1OriginRef", "block_id", l1OriginRefBlkID)
+	} else {
+		l1OriginRefBlkID = parentRef.L1Origin
+
+		b.log.Debug("Builder NewJob using L1Origin value from parentRef, as opts.L1Origin is nil", "block_id", l1OriginRefBlkID)
 	}
-	attrs, err := b.attrPrep.PreparePayloadAttributes(ctx, parentRef, l1OriginRef.ID())
+
+	attrs, err := b.attrPrep.PreparePayloadAttributes(ctx, parentRef, l1OriginRefBlkID)
 	if err != nil {
 		return nil, fmt.Errorf("failed to prepare payload attributes: %w", err)
 	}
-	id := seqtypes.RandomJobID()
-	info, err := b.cl.OpenBlock(ctx, parentRef.ID(), attrs)
+	b.log.Debug("Builder NewJob prepared payload attrs", "attrs", attrs)
+
+	address, err := randomAddress()
 	if err != nil {
-		return nil, fmt.Errorf("failed to open block: %w", err)
+		return nil, fmt.Errorf("failed to generate random address: %w", err)
 	}
+	attrs.SuggestedFeeRecipient = address
```

Comment:
This does not seem right: the L2 block must have the sequencer-fee-vault address as fee-recipient to be valid during consolidation. A random fee-recipient will always be invalid.